exporters/otlp/internal/transform/span_test.go

```diff
@@ -0,0 +1,399 @@
+		MessageEvents: []export.Event{
+			{Time: startTime,
+				Attributes: []core.KeyValue{
+					core.Key("CompressedByteSize").Uint64(512),
+				},
+			},
+			{Time: endTime,
+				Attributes: []core.KeyValue{
+					core.Key("MessageEventType").String("Recv"),
+				},
+			},
+		},
+		Links: []apitrace.Link{
+			{
+				SpanContext: core.SpanContext{
+					TraceID:    core.TraceID{0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF},
+					SpanID:     core.SpanID{0xB0, 0xB1, 0xB2, 0xB3, 0xB4, 0xB5, 0xB6, 0xB7},
+					TraceFlags: 0,
+				},
+				Attributes: []core.KeyValue{
+					core.Key("LinkType").String("Parent"),
+				},
+			},
+			{
+				SpanContext: core.SpanContext{
+					TraceID:    core.TraceID{0xE0, 0xE1, 0xE2, 0xE3, 0xE4, 0xE5, 0xE6, 0xE7, 0xE8, 0xE9, 0xEA, 0xEB, 0xEC, 0xED, 0xEE, 0xEF},
+					SpanID:     core.SpanID{0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0xD5, 0xD6, 0xD7},
+					TraceFlags: 0,
+				},
+				Attributes: []core.KeyValue{
+					core.Key("LinkType").String("Child"),
+				},
+			},
+		},
+		StatusCode:      codes.Internal,
+		StatusMessage:   "utterly unrecognized",
+		HasRemoteParent: true,
+		Attributes: []core.KeyValue{
+			core.Key("timeout_ns").Int64(12e9),
+		},
+		DroppedAttributeCount:    1,
+		DroppedMessageEventCount: 2,
+		DroppedLinkCount:         3,
+		Resource:                 resource.New(core.Key("rk1").String("rv1"), core.Key("rk2").Int64(5)),
+	}
+
+	// Not checking resource as the underlying map of our Resource makes
+	// ordering impossible to guarantee on the output. The Resource
+	// transform function has unit tests that should suffice.
+	expectedSpan := tracepb.Span{
+		TraceId:                []byte{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F},
+		SpanId:                 []byte{0xFF, 0xFE, 0xFD, 0xFC, 0xFB, 0xFA, 0xF9, 0xF8},
+		ParentSpanId:           []byte{0xEF, 0xEE, 0xED, 0xEC, 0xEB, 0xEA, 0xE9, 0xE8},
+		Name:                   spanData.Name,
+		Kind:                   tracepb.Span_SERVER,
+		StartTimeUnixNano:      uint64(startTime.Nanosecond()),
+		EndTimeUnixNano:        uint64(endTime.Nanosecond()),
+		Status:                 status(spanData.StatusCode, spanData.StatusMessage),
+		Events:                 spanEvents(spanData.MessageEvents),
+		Links:                  links(spanData.Links),
+		Attributes:             Attributes(spanData.Attributes),
+		DroppedAttributesCount: 1,
+		DroppedEventsCount:     2,
+		DroppedLinksCount:      3,
+	}
+
+	got := SpanData([]*export.SpanData{spanData})
+	if !assert.Len(t, got, 1) {
+		return
+	}
+
+	// Break the span down as large diffs can be hard to read.
+	actualSpans := got[0].GetInstrumentationLibrarySpans()
+	if !assert.Len(t, actualSpans, 1) && !assert.Len(t, actualSpans[0].Spans, 1) {
+		return
+	}
+	actualSpan := actualSpans[0].Spans[0]
+
+	if diff := cmp.Diff(expectedSpan.TraceId, actualSpan.TraceId, cmp.Comparer(proto.Equal)); diff != "" {
```

Comment:
Does comparing entire span not work?
```go
if diff := cmp.Diff(expectedSpan, actualSpan, cmp.Comparer(proto.Equal)); diff != "" {
     ...
}
```

Reply:
You're correct, this is a holdover from some debugging. I'll update.